ipfs-cluster-service/migrate.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bufio"
+
+	"fmt"
+	"os"
+
+	"github.com/urfave/cli"
+
+	peer "github.com/libp2p/go-libp2p-peer"
+	
+	ipfscluster "github.com/ipfs/ipfs-cluster"
+	"github.com/ipfs/ipfs-cluster/state/mapstate"
+	"github.com/ipfs/ipfs-cluster/consensus/raft"
+)
+
+
+func upgrade(c *cli.Context) error {
+	if c.NArg() < 1 || c.NArg() > 2 {
+		return fmt.Errorf("Usage: <BACKUP-FILE-PATH> [RAFT-DATA-DIR]")
+	}
+	//Load configs                                                             
+	cfg, clusterCfg, _, _, consensusCfg, _, _, _ := makeConfigs()
+	err := cfg.LoadJSONFromFile(configPath)
+	if err != nil {
+		return err
+	}
+	backupFilePath := c.Args().First()
+	var raftDataPath string
+	if c.NArg() == 1 {
+		raftDataPath = consensusCfg.DataFolder
+	} else {
+		raftDataPath = c.Args().Get(1)
+	}
+
+	//Migrate backup to new state                                                              
+	backup, err := os.Open(backupFilePath)
+	if err != nil {
+		return err
+	}
+
+	defer backup.Close()
+	r := bufio.NewReader(backup)
+	newState := mapstate.NewMapState()
+	err = newState.Restore(r)
+	if err != nil {
+		return err
+	}
+	//Record peers of cluster                                                                  
+	var peers []peer.ID
```

Comment:
if SnapshotReset does as explained above, it should be happy just with current peer ID. Also this is going to break with my PR since we are going to support multiple addresses per peer ID.